core/src/main/java/org/apache/carbondata/core/util/DeleteLoadFolders.java

```diff
@@ -115,7 +121,8 @@ private static void physicalFactAndMeasureMetadataDeletion(CarbonTable carbonTab
     SegmentUpdateStatusManager updateStatusManager =
         new SegmentUpdateStatusManager(carbonTable, currLoadDetails);
     for (final LoadMetadataDetails oneLoad : loadDetails) {
-      if (canDeleteThisLoad(oneLoad, isForceDelete, cleanStaleInProgress)) {
+      if (loadsToDelete.contains(oneLoad.getLoadName()) && canDeleteThisLoad(oneLoad,
```

Comment:
canDeleteThisLoad would be redundant here, as loadsToDelete is absolute

Reply:
removed